src/server/clientiface.cpp

```diff
@@ -338,7 +338,11 @@ void RemoteClient::GetNextBlocks (
 				*/
 				if (d >= d_opt && block->isAir())
 						continue;
+			} else if (emerge->isBlockInQueue(p)) {
+				// if the block is already in the emerge queue we don't have to check again
+				continue;
```

Comment:
I suppose this should still set `nearest_emerged_d`?

Reply:
That's a very good question... Hmm, does it?
I was thinking this just like when nothing is sent due to `blocks_sending` or `blocks_sent`. Now thinking about it again, I think you are right.
In fact, this should also if `block->isgenerated` is false;

Reply:
This even slightly faster since it covers the `!block->isGenerated()` case.